src/ansys/dpf/composites/sampling_point_solid_stack.py

```diff
@@ -0,0 +1,794 @@
+    """Implements the ``Sampling Point`` object for a stack of solid elements.
+
+    This class provides for plotting the lay-up and results at a certain point of the
+    layered structure. The results, including ``analysis_plies``, ``e1``, ``s12``, and
+    ``failure_modes``, are always from the bottom to the top of the laminate (along
+    the element normal direction). Postprocessing results such as ``e1`` are returned
+    as flat arrays where ``self.spots_per_ply`` can be used to compute the index for
+    a certain ply.
+
+    Parameters
+    ----------
+    name :
+        Name of the object.
+    element_id :
+       Element label of a solid element. The sampling will automatically select to
+       full stack of solid elements where this element is located.
+
+    Notes
+    -----
+    The results of layered elements are stored per integration point. A layered shell element
+    has a number of in-plane integration points (depending on the integration scheme) and
+    typically three integration points through the thickness. The through-the-thickness
+    integration points are called `spots`. They are typically at the ``BOTTOM``
+    and ``TOP`` of the layer. This notation is used here to identify the corresponding data.
+    Note that ``MIDDLE`` is not available for solid elements.
+
+    The ``SamplingPoint`` class returns three results per layer (one for each spot) because
+    the results of the in-plane integration points are interpolated to the centroid of the element.
+    The following table shows an example of a laminate with three layers. So a result, such as
+    ``s1`` has nine values, three for each ply.
+
+    +------------+------------+------------------------+
+    | Layer      | Index      | Spot                   |
+    +============+============+========================+
+    | Layer 3    | - 5        | - TOP of Layer 3       |
+    |            | - 4        | - BOTTOM of Layer 3    |
+    +------------+------------+------------------------+
+    | Layer 2    | - 3        | - TOP of Layer 2       |
+    |            | - 2        | - BOTTOM of Layer 2    |
+    +------------+------------+------------------------+
+    | Layer 1    | - 1        | - TOP of Layer 1       |
+    |            | - 0        | - BOTTOM of Layer 1    |
+    +------------+------------+------------------------+
+
+    The get_indices and get_offsets_by_spots methods simplify the indexing and
+    filtering of the data.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        element_id: int,
+        combined_criterion: CombinedFailureCriterion,
+        material_operators: MaterialOperators,
+        meshed_region: dpf.MeshedRegion,
+        layup_provider: dpf.Operator,
+        rst_streams_provider: dpf.Operator,
+        rst_data_source: dpf.DataSources,
+        element_info_provider: ElementInfoProvider,
+        default_unit_system: UnitSystem | None = None,
+        time: float | None = None,
```

Comment:
Would it make sense to make some of these arguments keyword-only (by adding a `*, ` to the arguments list)?

I think we have too many arguments here to reliably refer to them by position. The drawback is that (AFAICT) keyword-only arguments aren't used anywhere else in PyDPF Composites.